pennylane_lightning/core/src/simulators/lightning_gpu/initSV.cu

```diff
@@ -140,4 +247,41 @@ void setBasisState_CUDA(cuDoubleComplex *sv, cuDoubleComplex &value,
     setBasisState_CUDA_call(sv, value, index, async, stream_id);
 }
 
+void globalPhaseStateVector_CUDA(cuComplex *sv, size_t num_sv, cuComplex phase,
+                                 size_t thread_per_block,
+                                 cudaStream_t stream_id) {
+    globalPhaseStateVector_CUDA_call(sv, num_sv, phase, thread_per_block,
+                                     stream_id);
+}
+void globalPhaseStateVector_CUDA(cuDoubleComplex *sv, size_t num_sv,
+                                 cuDoubleComplex phase, size_t thread_per_block,
+                                 cudaStream_t stream_id) {
+    globalPhaseStateVector_CUDA_call(sv, num_sv, phase, thread_per_block,
+                                     stream_id);
+}
+
+void cGlobalPhaseStateVector_CUDA(cuComplex *sv, size_t num_sv, bool adjoint,
+                                  cuComplex *phase, size_t thread_per_block,
+                                  cudaStream_t stream_id) {
+    if (adjoint) {
+        cGlobalPhaseStateVector_CUDA_call<cuComplex, size_t, true>(
+            sv, num_sv, phase, thread_per_block, stream_id);
+    } else {
+        cGlobalPhaseStateVector_CUDA_call<cuComplex, size_t, false>(
+            sv, num_sv, phase, thread_per_block, stream_id);
+    }
+}
+void cGlobalPhaseStateVector_CUDA(cuDoubleComplex *sv, size_t num_sv,
+                                  bool adjoint, cuDoubleComplex *phase,
+                                  size_t thread_per_block,
+                                  cudaStream_t stream_id) {
+    if (adjoint) {
+        cGlobalPhaseStateVector_CUDA_call<cuDoubleComplex, size_t, true>(
+            sv, num_sv, phase, thread_per_block, stream_id);
+    } else {
+        cGlobalPhaseStateVector_CUDA_call<cuDoubleComplex, size_t, false>(
+            sv, num_sv, phase, thread_per_block, stream_id);
+    }
+}
+
```

Comment:
Not sure this file is the best overall for the location of these. Maybe we make an additional story to formalise the CUDA kernels when exploring the additions for the CPU-GPU box.